pippi_read.py

```diff
@@ -313,7 +313,7 @@ def getChainData(filename, cut_all_invalid=None, requested_cols=None, assignment
         data_isvalid.append(np.array(entries[column_names[index]+"_isvalid"], dtype=np.float64))
       lookup_key[index] = index_count
       index_count += 1
-    non_functional_cols = [i for i, elem in enumerate(data) if data[i] != 'functional']
+    non_functional_cols = [i for i, elem in enumerate(data) if np.all(data[i] != 'functional')] # Numpy now requires the user cast to a scalar when performing boolean checks on arrays
```

Comment:
This seems confusing, or at least not the optimal way to check.  What's being checked for here is just whether `data[i]` is a string with value 'functional', or a numpy array.  Better would probably be to just check if `data[i]` is a numpy array.  Same for the following changes using `np.all`.